.github/workflows/runIntegration.yml

```diff
@@ -0,0 +1,89 @@
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+        tags: true
+        token: ${{ secrets.FLYTE_BOT_PAT }}
+
+    - name: Log in to GitHub Container Registry
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ secrets.FLYTE_BOT_USERNAME }}
+        password: ${{ secrets.FLYTE_BOT_PAT }}
+
+    - name: Iterate through top level directories and build Docker images
+      env:
+        ACCESS_TOKEN: ${{ secrets.FLYTE_BOT_PAT }}
+      run: |
+            # Get the SHA of the base commit of the PR
+            base_sha=${{ github.event.pull_request.base.sha }}
+            special_tag=build-success-${{github.event.pull_request.number}}
+            # Get the list of changed files between the last commit and the current one
+            latest_tag=$(git tag -l $special_tag | sort -V | tail -n 1)
+            if [ -z "$latest_tag" ]; then
+              echo "no previous successful build tag exists"
+              # If no previous successful build tag exists, check all commits
+              changed_files=$(git diff --name-only $base_sha HEAD)
+            else
+              # Otherwise, check only the commits since the last successful build
+              echo "previous successful build tag exists"
+              changed_files=$(git diff --name-only $latest_tag HEAD)
+            fi
+            echo "Changed files: $changed_files"
+            for dir in *; do
+              if [ -f "$dir/{{cookiecutter.project_name}}/Dockerfile" ]; then
+                # Check if any of the changed files are in the current directory
+                if echo "$changed_files" | grep -q "^$dir/"; then
+                  echo "Building and pushing Docker image for $dir"
+                  docker build -f $dir/{{cookiecutter.project_name}}/Dockerfile -t ghcr.io/${{ github.repository }}:${dir}-pr-${{github.event.pull_request.number}} $dir/{{cookiecutter.project_name}}
+                  docker push ghcr.io/${{ github.repository }}:${dir}-pr-${{github.event.pull_request.number}}
+                else
+                   echo "Skipping $dir as it does not contain changes"
+                fi
+              else
+                echo "Skipping $dir as it does not contain a Dockerfile"
+              fi
+            done
+            if [ -z "$latest_tag" ]; then
+              echo "No previous successful build tag exists, so no need to delete it"
+            else
+              echo "Deleting last tag and replacing it with this one, since it was successful"
+              git tag -d $latest_tag
+              git push origin :refs/tags/$latest_tag
+            fi
+            git config user.name "GitHub Actions"
+            git config user.email "actions@github.com"
+            git tag build-success-${{ github.event.pull_request.number }} HEAD
+            git push origin build-success-${{ github.event.pull_request.number }}
+
+  integration_tests:
```

Comment:
Thanks for driving this! some ideas
1. we could have two GH workflows.
    - One submits the ML, spark, BQ workflows to union cloud that require external service or significant computing resources.
    - Another One runs the dynamic task, map task, regular python task in the flytectl demo cluster. 
2. we can have a workflow to test image spec. [Github action setup for envd](https://envd.tensorchord.ai/teams/ci.html#github-actions)
3. Use fast-register instead. I think is fine for testing. Building the image always take lots of time in the CI
4. Run the tests twice a day with latest flytekit or trigger the workflow once flytekit pr is merged

Reply:
1 - I think that makes a lot of sense, good split. However where does the flytectl demo cluster live?
2 - yup that's definitely something we should have, it'd be a template just like the others I think.
3 - Image is only rebuilt if the code changes, if it doesn't change it's skipped. But buildkit would definitely make this easier.
4 - yes exactly